test/strategy/ExtendConfigStrategyTest.js

```diff
@@ -7,6 +7,43 @@ var strategy = require('../../lib/strategy');
 var ExtendConfigStrategy = strategy.ExtendConfigStrategy;
 
 describe('ExtendConfigStrategy', function () {
+  it("should be able to extend member that references itself", function (done) {
+    var extendConfig = {
+      abc: "123"
+    };
+
+    extendConfig.root = extendConfig;
+    extendConfig.something = {
+      root: extendConfig
+    };
+
+    var strategy = new ExtendConfigStrategy(extendConfig);
+
+    strategy.process({}, function(err, config) {
+      if (err) {
+        throw err;
+      }
+
+      assert.deepEqual(config, {
+        abc: "123",
+        root: extendConfig,
+        something: {
+          root: extendConfig
+        }
+      });
+
+      // Assert that any circular references aren't broken.
+      var cursor = config;
+      for (var i = 0; i < 100; i++) {
```

Comment:
It's just a sanity check to see if it it can traverse it deep enough. 100 is within the sane limit of anything circular being traversed.